### PR TITLE
chore: Enzyme replacement in test files- PR 1 of 6 for part 3 of 3

### PR DESCRIPTION
**Pull request by v-singhanjal**
#### Details
Enzyme replacement in files under src/tests/unit/tests/DetailsView/
 

##### Motivation
Existing story: [User Story 2142857](https://dev.azure.com/mseng/1ES/_workitems/edit/2142857)
Replaced the use of Enzyme in the below files:

1. src/tests/unit/tests/DetailsView/components/status-icon.test.tsx
2. src/tests/unit/tests/DetailsView/components/switcher.test.tsx
3. src/tests/unit/tests/DetailsView/components/tab-stops-failed-instance-section.test.tsx
4. src/tests/unit/tests/DetailsView/components/tab-stops-minimal-requirement-header.test.tsx
5. src/tests/unit/tests/DetailsView/components/tab-stops-requirement-instances-collapsible-content.test.tsx
6. src/tests/unit/tests/DetailsView/components/tab-stops-requirements-with-instances.test.tsx
7. src/tests/unit/tests/DetailsView/components/target-change-dialog.test.tsx
8. src/tests/unit/tests/DetailsView/components/target-page-changed-view.test.tsx
9. src/tests/unit/tests/DetailsView/components/target-page-hidden-bar.test.tsx
10. src/tests/unit/tests/DetailsView/components/test-status-choice-group.test.tsx
11. src/tests/unit/tests/DetailsView/components/test-view-container.test.tsx
12. src/tests/unit/tests/DetailsView/components/transfer-to-assessment-button.test.tsx
13. src/tests/unit/tests/DetailsView/details-view-body.test.tsx
14. src/tests/unit/tests/DetailsView/details-view-container.test.tsx
15. src/tests/unit/tests/DetailsView/details-view-content.test.tsx
 

#### Pull request checklist

- [x] Addresses an existing issue:  [User Story 2142857](https://dev.azure.com/mseng/1ES/_workitems/edit/2142857)
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [n/a] (UI changes only) Added screenshots/GIFs to description above
- [n/a] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/tests/unit/tests/DetailsView/components/switcher.test.tsx [hunk 1]
```diff
@@ -24,72 +32,72 @@ describe('Switcher', () => {
...
     describe('props', () => {
         it('dropdown has correct options', () => {
-            const renderer = shallow(<Switcher {...defaultProps} />);
+            render(<Switcher {...defaultProps} />);
 
-            const dropdown = renderer.find(Dropdown);
+            const dropdown = getMockComponentClassPropsForCall(Dropdown);
 
-            expect(dropdown.prop('options')).toMatchSnapshot();
+            expect(dropdown).toMatchSnapshot();
```
Comment (v-rakeshsh) [inline src/tests/unit/tests/DetailsView/components/switcher.test.tsx:59]: hope this dropdown has options props?

File: src/tests/unit/tests/DetailsView/components/tab-stops-requirement-instances-collapsible-content.test.tsx [hunk 2]
```diff
@@ -48,41 +58,44 @@ describe('TabStopsRequirementInstancesCollapsibleContent', () => {
...
         );
-        testSubject
-            .find(Link)
-            .find('button')
-            .forEach(wrapper => wrapper.simulate('click'));
+
+        const button = renderResult.getAllByRole('button');
+
+        await userEvent.click(button[0]);
+        await userEvent.click(button[1]);
```
[the remark this answers is not in the document]
Comment (v-singhanjal) [reply src/tests/unit/tests/DetailsView/components/tab-stops-requirement-instances-collapsible-content.test.tsx:98]: There was a forEach loop here. The loop was not working for the 2 buttons so added separately for each button.

File: src/tests/unit/tests/DetailsView/components/switcher.test.tsx (hunk 1 above)
Comment (v-singhanjal) [reply src/tests/unit/tests/DetailsView/components/switcher.test.tsx]: Made the change

File: src/tests/unit/tests/DetailsView/details-view-container.test.tsx [hunk 3]
```diff
@@ -99,11 +113,15 @@ describe('DetailsViewContainer', () => {
...
-            expect(rendered.getElement()).toMatchSnapshot();
+            const renderResult = render(<DetailsViewContainer {...props} />);
+            expect(renderResult.asFragment()).toMatchSnapshot();
+            expectMockedComponentPropsToMatchSnapshots([NarrowModeDetector]);
         });
 
         it('render once; should call details view opened', () => {
+            useOriginalReactElements('DetailsView/actions/details-view-action-message-creator', [
+                'DetailsViewActionMessageCreator',
+            ]);
```
Comment (madalynrose) [inline src/tests/unit/tests/DetailsView/details-view-container.test.tsx:124]: This isn't actually needed
```suggestion
```

File: src/tests/unit/tests/DetailsView/details-view-container.test.tsx [hunk 4]
```diff
@@ -129,12 +147,16 @@ describe('DetailsViewContainer', () => {
...
+            const renderResult = render(<DetailsViewContainer {...props} />);
+            expect(renderResult.asFragment()).toMatchSnapshot();
+            expectMockedComponentPropsToMatchSnapshots([NarrowModeDetector]);
             detailsViewActionMessageCreator.verifyAll();
         });
 
         it('render twice; should not call details view opened on second render', () => {
+            useOriginalReactElements('DetailsView/actions/details-view-action-message-creator', [
+                'DetailsViewActionMessageCreator',
+            ]);
```
Comment (madalynrose) [inline src/tests/unit/tests/DetailsView/details-view-container.test.tsx:159]:
```suggestion
```

File: src/tests/unit/tests/DetailsView/details-view-container.test.tsx [hunk 5]
```diff
@@ -8,11 +10,20 @@ import {
...
-import { shallow } from 'enzyme';
 import * as React from 'react';
 import { IMock, Mock, Times } from 'typemoq';
-
+import { NarrowModeDetector } from '../../../../DetailsView/components/narrow-mode-detector';
+import { NoContentAvailable } from '../../../../DetailsView/components/no-content-available/no-content-available';
+import {
+    expectMockedComponentPropsToMatchSnapshots,
+    mockReactComponents,
+    useOriginalReactElements,
```
Comment (madalynrose) [inline src/tests/unit/tests/DetailsView/details-view-container.test.tsx:20]: We end up not needing this.
```suggestion
```

**Review by madalynrose — CHANGES_REQUESTED** [review]
This looks awesome! There were a lot of complex tests here. I made some suggestions and once those are implemented it's ready to go!

**Review by madalynrose — APPROVED** [review]

**Pull request merged**